ydb/docs/en/core/development/localdb-uncommitted-txs.md

```diff
@@ -0,0 +1,178 @@
+| ... | ... |
+| offset of Main | header of Main |
+
+Having a pointer to Delta 0, other records for the same key are available with the `GetAltRecord(size_t index)` method, where `index` is the record number (which is 1 for Delta 1). The chain of records ends either with a pointer to the record without an IsDelta flag (the Main record), or 0 (when there is no Main record for the key).
+
+Let's suppose that after writing tx 13 above the MemTable was compacted. Entry for the 32-bit key K may look like this (offsets are relative to the record pointer on the table):
+
+| Offset | Value | Description |
+--- | --- | ---
+| -16 | 58 | offset of Main |
+| -8 | 29 | offset of Delta 1 |
+| 0 | 0x21 | Delta 0: IsDelta + ERowOp::Upsert |
+| 1 | 0x00 | .. key column is not NULL |
+| 2 | K | .. key column (32-bit) |
+| 6 | 0x00 | .. column A is empty |
+| 7 | 0 | .. column A (32-bit) |
+| 11 | 0x01 | .. column B = ECellOp::Set |
+| 12 | 20 | .. column B (32-bit) |
+| 16 | 0x00 | .. column C is empty |
+| 17 | 0 | .. column C (32-bit) |
+| 21 | 13 | .. TDelta::TxId |
+| 29 | 0x21 | Delta 1: IsDelta + ERowOp::Upsert |
+| 30 | 0x00 | .. key column is not NULL |
+| 31 | K | .. key column (32-bit) |
+| 35 | 0x00 | .. column A is empty |
+| 36 | 0 | .. column A (32-bit) |
+| 40 | 0x00 | .. column B is empty |
+| 41 | 0 | .. column B (32-bit) |
+| 45 | 0x01 | .. column C = ECellOp::Set |
+| 46 | 10 | .. column C (32-bit) |
+| 50 | 15 | .. TDelta::TxId |
+| 58 | 0x61 | Main: HasHistory + IsVersioned + ERowOp::Upsert |
+| 59 | 0x00 | .. key column is not NULL |
+| 60 | K | .. key column (32-bit) |
+| 64 | 0x01 | .. column A = ECellOp::Set |
+| 65 | 1 | .. column A (32-bit) |
+| 69 | 0x01 | .. column B = ECellOp::Set |
+| 70 | 2 | .. column B (32-bit) |
+| 74 | 0x01 | .. column C = ECellOp::Set |
+| 75 | 3 | .. column C (32-bit) |
+| 79 | 3000 | .. RowVersion.Step |
+| 87 | 12 | .. RowVersion.TxId |
+| 95 | - | End of record |
+
+The HasHistory flag in the Main record shows that other two records are stored among history data with keys `(RowId, 2000, 11)` and `(RowId, 1000, 10)` respectively.
+
+When compacting iterator runs in a special mode that enumerates all deltas and all versions for each key. The compaction scan implementation (see [flat_ops_compact.h](https://github.com/ydb-platform/ydb/blob/main/ydb/core/tablet_flat/flat_ops_compact.h)) first aggregates all uncommitted deltas by their TxId in the same order (in case changes from different TxIds overlap their order may change arbitrarily, which is OK since such transactions are not supposed to both commit). After uncommitted deltas are aggregated, they a flushed to the resulting SST (see [flat_part_writer.h](https://github.com/ydb-platform/ydb/blob/main/ydb/core/tablet_flat/flat_part_writer.h) and [flat_page_writer.h](https://github.com/ydb-platform/ydb/blob/main/ydb/core/tablet_flat/flat_page_writer.h)), and committed row versions for the same key are enumerated, which are written in decreasing version order.
```

Comment:
```suggestion
When compacting iterator runs in a special mode that enumerates all deltas and all versions for each key. The compaction scan implementation (see [flat_ops_compact.h](https://github.com/ydb-platform/ydb/blob/main/ydb/core/tablet_flat/flat_ops_compact.h)) first aggregates all uncommitted deltas by their TxId in the same order (in case changes from different TxIds overlap their order may change arbitrarily, which is OK since such transactions are not supposed to both commit). After uncommitted deltas are aggregated, they are flushed to the resulting SST (see [flat_part_writer.h](https://github.com/ydb-platform/ydb/blob/main/ydb/core/tablet_flat/flat_part_writer.h) and [flat_page_writer.h](https://github.com/ydb-platform/ydb/blob/main/ydb/core/tablet_flat/flat_page_writer.h)), and committed row versions for the same key are enumerated, which are written in decreasing version order.
```